argilla-server/src/argilla_server/_app.py

```diff
@@ -121,6 +124,32 @@ def configure_api_router(app: FastAPI):
     app.mount("/api/v1", api_v1)
 
 
+def configure_telemetry(app: FastAPI):
+    """
+    Configures telemetry middleware for the app if telemetry is enabled
+
+    """
+    if not settings.enable_telemetry:
+        return
+
+    @app.middleware("http")
+    async def track_api_endpoint(request: Request, call_next):
```

Comment:
I believe a better naming could be something like:

```suggestion
    async def track_api_request(request: Request, call_next):
```